W-17017575-insights-hyperforce-duke (#314)
mulesoft/docs-monitoring: modules/ROOT/pages/logs.adoc

====

[[log-search]]
== Search Logs

> From Anypoint Monitoring, find *Log Search* to search for text in logs from the search field, through a filter, or with a DSL query.

Reviewer: ```suggestion
From Anypoint Monitoring, click *Log Search* to search for text in logs from the search field, through a filter, or with a DSL query.
```

Reviewer: (Not sure if that's correct.)

image::intro-logs.png["Logs in Anypoint Monitoring"]

* <<search-field>>
* <<search-filter>>
* <<search-dsl>>

When you search for logs for APIs deployed through Flex Gateway, use the *Application* filter. Use the *App ID* field to identify the instance.

[[search-field]]
=== Search from the Log Search Field

To search the logs, type your search query into the log search field and press *Enter*. 

> * Search for a term, such as `"Info Dynamic Logging"`.

Reviewer: I think, in this case, just the "" is necessary (no font change).

Author: ```suggestion
* Search for a term, such as "Info Dynamic Logging".
```

Author: correcting

Author: ok as-is, based on comment below

+

> Without quotes, the query returns results for matches to each term. 

Reviewer: Ooooh, nm.

> * Use a wildcard in your search, such as `resourceId=hellowor*. 

Reviewer: ```suggestion
* Use a wildcard in your search, such as `resourceId=hellowor*`. 
```

This example of a predefined filter selects the ERROR log level.

image::log-filter-select.png["Predefined log filter"]

After you select a predefined filter from the Quick Filter list, you see it
near the top of the page, next to *Add a Filter*.

> image::log-filter-actions.png["Filter actions"]

Reviewer: You could give more context to these alt texts - maybe "Predefined filter actions on the xxx page"

Author: ```suggestion
image::log-filter-actions.png["Predefined filter actions"]
```

The resulting `log-level` column looks like this in the logs:

image::log-filter-column.png["Filter column in the logs"]

To remove the filter column, click the column icon for the filter field,
for example:

> image::log-filter-remove-field.png["Removing a filter column from the logs"]

Reviewer: Instead of these gerund constructions, maybe make it more descriptive - Filter field column icon

Author: ```suggestion
image::log-filter-remove-field.png["Column icon for removing the column from the logs"]
```

To move filter columns in your log results to the left or right, you simply scroll over the column heading and use the arrow icons.

> image::log-filter-column-move.png["Removing a filter column from the logs"]

Author: ```suggestion
image::log-filter-column-move.png["Arrow icons for moving a filter column to the left or right"]
```

You can use the funnel-shaped icons on any log in your results to create and remove a filter for the entire log field value.

> image::log-filter-create-funnel.png["Creating a log message filter"]

Author: ```suggestion
image::log-filter-create-funnel.png["Funnel icon used to create a log message filter"]
```
